### PR TITLE
SIRI-88 Codelisteneinträge Import

### PR DESCRIPTION
**Pull request by MGrammet**
Ermöglicht den Import von Codelisteneinträgen.

fixes SIRI-88

### CODE REVIEW

File: src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ * Provides an import job for {@link SQLCodeList code lists} stored in a JDBC database.
+ */
+@Register(classes = JobFactory.class, framework = SQLTenants.FRAMEWORK_TENANTS_JDBC)
+@Permission(CodeListController.PERMISSION_MANAGE_CODELISTS)
+public class SQLCodeListEntryImportJobFactory extends LineBasedImportJobFactory {
+
+    /**
+     * Contains the sql code list to import the code list entries into.
+     */
+    private SQLCodeListParameter codeListParameter = new SQLCodeListParameter("SQLCodeList");
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java]: $CodeList?

File: src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            codeList = process.require(codeListParameter);
+        }
+
+        @Override
+        protected SQLCodeListEntry findAndLoad(Context ctx) {
+            ctx.put(SQLCodeListEntry.CODE_LIST.toString(), codeList.getId());
+            return super.findAndLoad(ctx);
+        }
+
+        /**
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java]: don't copy javadoc for overwritten methods if there is nothing to add

File: src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    @Override
+    protected Class<? extends BaseEntity<?>> getImportType() {
+        return SQLCodeListEntry.class;
+    }
+
+    @Override
+    protected boolean hasPresetFor(Object targetObject) {
+        return targetObject == SQLCodeListEntry.class;
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java]: SQLCodeList?

Comment (andyHa) [reply src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java]: computePreset?

File: src/main/java/sirius/biz/jobs/params/CodeListParameter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public CodeListParameter(String name, String label) {
+        super(name, label);
+    }
+
+    /**
+     * Creates a new parameter with the given name.
+     *
+     * @param name the name of the parameter
+     */
+    public CodeListParameter(String name) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/jobs/params/CodeListParameter.java]: protected for abstract base classes - please execut sonarlint at least once!

File: src/main/java/sirius/biz/jobs/params/CodeListParameter.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+public abstract class CodeListParameter<I, L extends BaseEntity<I> & CodeList>
+        extends EntityParameter<L, CodeListParameter<I, L>> {
+
+    /**
+     * Creates a new parameter with the given name and label.
+     *
+     * @param name  the name of the parameter
+     * @param label the label of the parameter, which will be {@link NLS#smartGet(String) auto translated}
+     */
+    public CodeListParameter(String name, String label) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/jobs/params/CodeListParameter.java]: s.b.

File: src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportJobFactory.java]: codeList, "$CcodeList"

File: src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportHandler.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Value;
+import sirius.kernel.di.std.Register;
+
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Provides an import handler for {@link SQLCodeListEntry code lists}.
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/jdbc/SQLCodeListEntryImportHandler.java]:
```suggestion
 * Provides an import handler for {@link SQLCodeListEntry code list entries}.
```

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 7]
```diff
@@ -189,4 +190,24 @@ public void deleteCodeListEntry(WebContext ctx, String codeListId, String entryI
...
     }
+
+    /**
+     * Autocompletion for codelists.
+     *
+     * @param ctx the current request
+     */
+    @LoginRequired
+    @Routed("/code-lists/autocomplete")
+    public void codeListsAutocomplete(final WebContext ctx) {
```
Comment (sabieber) [inline src/main/java/sirius/biz/codelists/CodeListController.java]:
```suggestion
    public void codeListsAutocomplete(final WebContext webContext) {
```

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 8]
```diff
@@ -189,4 +190,24 @@ public void deleteCodeListEntry(WebContext ctx, String codeListId, String entryI
...
+        AutocompleteHelper.handle(ctx, (query, result) -> {
```
Comment (sabieber) [inline src/main/java/sirius/biz/codelists/CodeListController.java]:
```suggestion
        AutocompleteHelper.handle(webContext, (query, result) -> {
```

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 9]
```diff
@@ -189,4 +190,24 @@ public void deleteCodeListEntry(WebContext ctx, String codeListId, String entryI
...
+            BasePageHelper<L, ?, ?, ?> ph = getListsAsPage();
```
Comment (sabieber) [inline src/main/java/sirius/biz/codelists/CodeListController.java]:
```suggestion
            BasePageHelper<L, ?, ?, ?> pageHelper = getListsAsPage();
```

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 10]
```diff
@@ -189,4 +190,24 @@ public void deleteCodeListEntry(WebContext ctx, String codeListId, String entryI
...
+            ph.withContext(ctx);
```
Comment (sabieber) [inline src/main/java/sirius/biz/codelists/CodeListController.java]:
```suggestion
            pageHelper.withContext(webContext);
```

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 11]
```diff
@@ -189,4 +190,24 @@ public void deleteCodeListEntry(WebContext ctx, String codeListId, String entryI
...
+
+            ph.asPage().getItems().forEach(codeList -> {
```
Comment (sabieber) [inline src/main/java/sirius/biz/codelists/CodeListController.java]:
```suggestion
            pageHelper.asPage().getItems().forEach(codeList -> {
```

**Pull request merged**